app/models/miq_user_role.rb

```diff
@@ -141,8 +141,7 @@ def self.display_name(number = 1)
   private
 
   def restriction_type(klass)
-    case klass.to_s
-    when "ServiceTemplate"
+    if klass <= ServiceTemplate
```

Comment:
I'm looking at this more closely and believe this needs to handle `nil` klass value. Moving to WIP

Reply:
> can you use `klass.kind_of?(ServiceTemplate)` ?

Doesn't work, I think because `klass` is a class object, not an instance of `ServiceTemplate` type:
```ruby
(byebug) klass
ServiceTemplate(id: integer, name: string, description: string, guid: string, type: string, service_template_id: integer, options: text, created_at: datetime, updated_at: datetime, display: boolean, evm_owner_id: integer, miq_group_id: integer, service_type: string, prov_type: string, provision_cost: float, service_template_catalog_id: integer, long_description: text, tenant_id: integer, generic_subtype: string, deleted_on: datetime, internal: boolean, zone_id: integer, currency_id: integer, price: float, href_slug: string, region_number: integer, region_description: string, owned_by_current_user: boolean, owned_by_current_ldap_group: boolean, type_display: string, template_valid: boolean, template_valid_error_message: string, archived: boolean, active: boolean, evm_owner_email: string, evm_owner_name: string, evm_owner_userid: string, owning_ldap_group: string)
```

Reply:
Yeah, it's funny how it's easier to pass around objects than classes.

Reply:
Using `klass ||= Class` will cover klass = `nil` and `""`. I think it's safe to use `Class` as the default type that's covered in the `else` clause.

Reply:
> What do you think about a simple if klass && klass <= ServiceTemplate ?

Looks good to me.

> Since we aren't using klass later on we only need to check that it is non-nil in the conditional

I'd like to support additional restriction types in the near future - so I'm working with that in mind. With an additional type the `klass ||= Class` seems better. For example:
```ruby
  def restriction_type(klass)
    klass ||= Class
    if klass <= ServiceTemplate
      :service_templates
    elsif klass <= OrchestrationStack
      :orchestration_stacks
    else
      :vms
    end
  end
```

... of course I could stick with the simpler code today, and refactor when when support for additional restriction types is added.

Reply:
:+1: makes sense thanks